File: web/src/types/agent.ts
export interface AgentSessionSummary {
  id: number
  userId: number
  title: string
  bugTarget?: Record<string, unknown>
  pendingAction?: Record<string, unknown>
  createdAt?: string | null
  updatedAt?: string | null
}

export interface AgentMessage {
  id: number
  role: 'user' | 'assistant' | 'dispatch' | 'result' | string
  content: string
  dispatchId?: number | null
  attachments?: AgentAttachments
  createdAt?: string | null
}

export interface AgentImageAttachment {
  url: string
  thumbnailUrl?: string | null
  filename?: string
  mime?: string
  size?: number
}

export type AgentContextMode = 'standard' | 'quick'

export interface AgentContextRef {
  mode: AgentContextMode
  requirementItemId?: number | null
  quickSessionId?: string | null
  useCurrentFunctionMap?: boolean
}

export interface AgentAttachments {
  images?: AgentImageAttachment[]
  keyImages?: { image: string; platform?: string; index?: number; caption?: string }[]
  reportUrl?: string | null
  bugUrl?: string | null
  toolKey?: string | null
  status?: string | null
  [key: string]: unknown
}

export interface AgentSessionDetail {
  session: AgentSessionSummary
  messages: AgentMessage[]
}

export interface AgentUploadResult {
  images: AgentImageAttachment[]
}

File: backend/app/schemas/agent.py
from __future__ import annotations

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


class AgentImageAttachment(BaseModel):
    url: str
    thumbnail_url: str | None = None
    filename: str = ""
    mime: str = ""
    size: int = 0


class AgentMessageOut(BaseModel):
    id: int
    role: str
    content: str
    dispatch_id: int | None = None
    attachments: dict[str, Any] = Field(default_factory=dict)
    created_at: datetime | None = None

    model_config = {"from_attributes": True}


class AgentSessionSummaryOut(BaseModel):
    id: int
    user_id: int
    title: str
    bug_target: dict[str, Any] = Field(default_factory=dict)
    pending_action: dict[str, Any] = Field(default_factory=dict)
    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class AgentSessionOut(BaseModel):
    session: AgentSessionSummaryOut
    messages: list[AgentMessageOut] = Field(default_factory=list)


class AgentContextRef(BaseModel):
    mode: Literal["standard", "quick"]
    requirement_item_id: int | None = None
    quick_session_id: str | None = None
    use_current_function_map: bool = True


class AgentMessageCreateIn(BaseModel):
    content: str
    attachments: dict[str, Any] = Field(default_factory=dict)
    context_ref: AgentContextRef | None = None


class AgentUploadOut(BaseModel):
    images: list[AgentImageAttachment] = Field(default_factory=list)


class AgentCallbackOut(BaseModel):
    handled: bool
    event: str
    dispatch_id: int | None = None
    submission_id: str | None = None


class AgentToolListOut(BaseModel):
    tools: list[dict[str, Any]] = Field(default_factory=list)
